e2e/scenario/vhd.go

```
@@ -97,6 +97,28 @@ func mustGetVHDCatalogFromEmbeddedJSON(rawJSON string) VHDCatalog {
 	return catalog
 }
 
+type Manifest struct {
+	Containerd struct {
+		Edge string `json:"edge"`
+	} `json:"containerd"`
+}
+
+func getVHDManifest() (Manifest, error) {
+	manifestData, err := os.ReadFile("../parts/linux/cloud-init/artifacts/manifest.json")
+	if err != nil {
+		return Manifest{}, err
+	}
+	manifestDataStr := string(manifestData)
+	manifestDataStr = strings.TrimRight(manifestDataStr, "#EOF \n\r\t")
+	manifestData = []byte(manifestDataStr)
+
+	manifest := Manifest{}
+	if err = json.Unmarshal([]byte(manifestData), &manifest); err != nil {
+		panic(err)
```

lets just return the error here and panic in the parent call